### ISSUE TITLE
Cannot reproduce the results of Table 2

### ISSUE THREAD

**Issue by JunhyunB** [Weixin-Liang/MetaShift#4]
Thank you for sharing the MetaShift dataset.

However, I have some trouble reproducing the results of Table 2 (Subpopulation shift).

For p=6%, ERM shows 83.507% average accuracy and 73.6% worst-case accuracy (for iteration 40) which are much higher than the results reported in Table 2.

Also, for other algorithms, every result of Table 2 is not matched with my own results.

Is this because of the version of my python packages?

```
Environment:
	Python: 3.8.11
	PyTorch: 1.7.1
	Torchvision: 0.8.2
	CUDA: 11.0
	CUDNN: 8005
	NumPy: 1.20.3
	PIL: 8.3.1
Args:
	algorithm: ERM
	batch_size: 32
	data: MetaShift/dataset/data/MetaShift/MetaShift-subpopulation-shift
	hparams: None
	hparams_seed: 0
	log_prefix: 
	num_classes: 2
	num_domains: 2
	output_dir: train_output
	save_model_every_checkpoint: False
	seed: 0
	skip_model_save: False
	workers: 4
train_dataset.samples reverse: [('cat(indoor)', 800), ('dog(outdoor)', 800), ('cat(outdoor)', 50), ('dog(indoor)', 50)]
self.domain_to_groups {0: {'cat': ['cat(indoor)'], 'dog': ['dog(indoor)']}, 1: {'cat': ['cat(outdoor)'], 'dog': ['dog(outdoor)']}}
HParams:
	batch_size: 32
	class_balanced: False
	data_augmentation: True
	lr: 5e-05
	nonlinear_classifier: False
	resnet18: True
	resnet_dropout: 0.0
	weight_decay: 0.0
step_vals {'loss': 0.7582564353942871}
Iteration: 0
out-of-domain val
accuracy 0.543 	 roc_auc_score 0.693
confusion_matrix
[[ 31 257]
 [  6 282]]
classification_report
              precision    recall  f1-score   support

           0       0.84      0.11      0.19       288
           1       0.52      0.98      0.68       288

    accuracy                           0.54       576
   macro avg       0.68      0.54      0.44       576
weighted avg       0.68      0.54      0.44       576

VAL * Acc@1 54.340
 * Acc@1 54.340 Acc@5 0.000
accuracy 0.993 	 size: 144 	 dog(indoor)
accuracy 0.965 	 size: 144 	 dog(outdoor)
accuracy 0.118 	 size: 144 	 cat(indoor)
accuracy 0.097 	 size: 144 	 cat(outdoor)
step_vals {'loss': 0.8029559254646301}
step_vals {'loss': 0.6127275228500366}
step_vals {'loss': 0.5663400888442993}
step_vals {'loss': 0.5361894965171814}
step_vals {'loss': 0.5080360770225525}
step_vals {'loss': 0.502351701259613}
step_vals {'loss': 0.45667174458503723}
step_vals {'loss': 0.4510887563228607}
step_vals {'loss': 0.4313385784626007}
step_vals {'loss': 0.41102027893066406}
step_vals {'loss': 0.4396398067474365}
step_vals {'loss': 0.35080477595329285}
step_vals {'loss': 0.3561045825481415}
step_vals {'loss': 0.4412407875061035}
step_vals {'loss': 0.3722797632217407}
step_vals {'loss': 0.40952974557876587}
step_vals {'loss': 0.3348669111728668}
step_vals {'loss': 0.28301629424095154}
step_vals {'loss': 0.2869456112384796}
step_vals {'loss': 0.2969983220100403}
Iteration: 20
out-of-domain val
accuracy 0.835 	 roc_auc_score 0.913
confusion_matrix
[[250  38]
 [ 57 231]]
classification_report
              precision    recall  f1-score   support

           0       0.81      0.87      0.84       288
           1       0.86      0.80      0.83       288

    accuracy                           0.84       576
   macro avg       0.84      0.84      0.83       576
weighted avg       0.84      0.84      0.83       576

VAL * Acc@1 83.507
 * Acc@1 83.507 Acc@5 0.000
accuracy 0.931 	 size: 144 	 cat(indoor)
accuracy 0.896 	 size: 144 	 dog(outdoor)
accuracy 0.806 	 size: 144 	 cat(outdoor)
accuracy 0.708 	 size: 144 	 dog(indoor)
step_vals {'loss': 0.27039286494255066}
step_vals {'loss': 0.30204248428344727}
step_vals {'loss': 0.21312294900417328}
step_vals {'loss': 0.2437962144613266}
step_vals {'loss': 0.25198787450790405}
step_vals {'loss': 0.23842911422252655}
step_vals {'loss': 0.1691025346517563}
step_vals {'loss': 0.24762478470802307}
step_vals {'loss': 0.28251034021377563}
step_vals {'loss': 0.14738459885120392}
step_vals {'loss': 0.18546883761882782}
step_vals {'loss': 0.22910372912883759}
step_vals {'loss': 0.21566125750541687}
step_vals {'loss': 0.1549062728881836}
step_vals {'loss': 0.13150382041931152}
step_vals {'loss': 0.1629757136106491}
step_vals {'loss': 0.21351677179336548}
step_vals {'loss': 0.1662304550409317}
step_vals {'loss': 0.13489097356796265}
step_vals {'loss': 0.15391628444194794}
Iteration: 40
out-of-domain val
accuracy 0.835 	 roc_auc_score 0.905
confusion_matrix
[[239  49]
 [ 46 242]]
classification_report
              precision    recall  f1-score   support

           0       0.84      0.83      0.83       288
           1       0.83      0.84      0.84       288

    accuracy                           0.84       576
   macro avg       0.84      0.84      0.84       576
weighted avg       0.84      0.84      0.84       576

VAL * Acc@1 83.507
 * Acc@1 83.507 Acc@5 0.000
accuracy 0.944 	 size: 144 	 dog(outdoor)
accuracy 0.924 	 size: 144 	 cat(indoor)
accuracy 0.736 	 size: 144 	 cat(outdoor)
accuracy 0.736 	 size: 144 	 dog(indoor)
step_vals {'loss': 0.09984487295150757}
step_vals {'loss': 0.12095910310745239}
step_vals {'loss': 0.15282633900642395}
step_vals {'loss': 0.24499759078025818}
step_vals {'loss': 0.17233917117118835}
step_vals {'loss': 0.1805627942085266}
step_vals {'loss': 0.29616454243659973}
step_vals {'loss': 0.18791015446186066}
step_vals {'loss': 0.13349810242652893}
step_vals {'loss': 0.17763575911521912}
step_vals {'loss': 0.14355479180812836}
step_vals {'loss': 0.2498779296875}
step_vals {'loss': 0.08334649354219437}
step_vals {'loss': 0.23552735149860382}
step_vals {'loss': 0.1597958207130432}
step_vals {'loss': 0.1945687234401703}
step_vals {'loss': 0.12305081635713577}
step_vals {'loss': 0.18060629069805145}
step_vals {'loss': 0.10387007147073746}
step_vals {'loss': 0.18083493411540985}
Iteration: 60
out-of-domain val
accuracy 0.823 	 roc_auc_score 0.910
confusion_matrix
[[232  56]
 [ 46 242]]
classification_report
              precision    recall  f1-score   support

           0       0.83      0.81      0.82       288
           1       0.81      0.84      0.83       288

    accuracy                           0.82       576
   macro avg       0.82      0.82      0.82       576
weighted avg       0.82      0.82      0.82       576

VAL * Acc@1 82.292
 * Acc@1 82.292 Acc@5 0.000
accuracy 0.965 	 size: 144 	 dog(outdoor)
accuracy 0.903 	 size: 144 	 cat(indoor)
accuracy 0.715 	 size: 144 	 dog(indoor)
accuracy 0.708 	 size: 144 	 cat(outdoor)
step_vals {'loss': 0.21658587455749512}
step_vals {'loss': 0.21305663883686066}
step_vals {'loss': 0.10273241996765137}
step_vals {'loss': 0.08029960840940475}
step_vals {'loss': 0.1491691768169403}
step_vals {'loss': 0.18964910507202148}
step_vals {'loss': 0.13241422176361084}
step_vals {'loss': 0.1288807988166809}
step_vals {'loss': 0.11977404356002808}
step_vals {'loss': 0.09882873296737671}
step_vals {'loss': 0.15328539907932281}
step_vals {'loss': 0.18410827219486237}
step_vals {'loss': 0.17586088180541992}
step_vals {'loss': 0.08180069923400879}
step_vals {'loss': 0.12042209506034851}
step_vals {'loss': 0.15169812738895416}
step_vals {'loss': 0.1118776723742485}
step_vals {'loss': 0.10913677513599396}
step_vals {'loss': 0.078999362885952}
step_vals {'loss': 0.17687441408634186}
Iteration: 80
out-of-domain val
accuracy 0.819 	 roc_auc_score 0.905
confusion_matrix
[[236  52]
 [ 52 236]]
classification_report
              precision    recall  f1-score   support

           0       0.82      0.82      0.82       288
           1       0.82      0.82      0.82       288

    accuracy                           0.82       576
   macro avg       0.82      0.82      0.82       576
weighted avg       0.82      0.82      0.82       576

VAL * Acc@1 81.944
 * Acc@1 81.944 Acc@5 0.000
accuracy 0.965 	 size: 144 	 dog(outdoor)
accuracy 0.951 	 size: 144 	 cat(indoor)
accuracy 0.688 	 size: 144 	 cat(outdoor)
accuracy 0.674 	 size: 144 	 dog(indoor)
```

**Comment by Weixin-Liang** [comment]
Thank you for your interest in our work! 

Please be advised that before making MetaShift public, we have made further efforts to reduce the label errors propagated from Visual Genome. Therefore, we expect a slight change in the exact experiment numbers. For example, Table 6 in ["Improving Out-of-Distribution Robustness via Selective Augmentation"](https://arxiv.org/pdf/2201.00299.pdf) uses a more update-to-date version of MetaShift.
